ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -58,56 +59,108 @@ func NewReadWriter() (*YAMLReadWriter, error) {
 	return &YAMLReadWriter{destination: dest}, nil
 }
 
-func readINI(dest *Destination) (*CLIConfig, map[interface{}]interface{}, error) {
+func readINI(dest *Destination) (*CLIConfig, error) {
 	iniReadWriter, err := NewINIReadWriter(dest)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	return iniReadWriter.GetConfig()
 }
 
-func readYAML(yamlPath string, configMap map[interface{}]interface{}, cliConfig *CLIConfig) error {
-	// convert yaml to CliConfig
-	dat, err := ioutil.ReadFile(yamlPath)
+// readClusterConfig does all the work to read and parse the yaml cluster config
+func readClusterConfig(path string, clusterConfigKey string, cliConfig *CLIConfig) error {
+	// read cluster file
+	dat, err := ioutil.ReadFile(path)
+	if err != nil {
+		return errors.Wrap(err, "Failed to read config file: "+path)
+	}
+
+	config := ClusterConfig{Clusters: make(map[string]Cluster)}
+	if err = yaml.Unmarshal(dat, &config); err != nil {
+		return errors.Wrap(err, "Failed to parse yaml file: "+path)
+	}
+
+	// get the correct cluster
+	chosenCluster := ""
+	if clusterConfigKey == "" {
+		chosenCluster = config.Default
+	} else {
+		chosenCluster = clusterConfigKey
+	}
+	cluster := config.Clusters[chosenCluster]
+
+	// Get the info out of the cluster
+	cliConfig.Region = cluster.Region
+	cliConfig.Cluster = cluster.Cluster
+	cliConfig.ComposeServiceNamePrefix = cluster.ComposeServiceNamePrefix
+	return nil
+
+}
+
+// readClusterConfig does all the work to read and parse the yaml cluster config
+func readProfileConfig(path string, profileConfigKey string, cliConfig *CLIConfig) error {
+	// read cluster file
```

Comment:
So the code is very similar, but its not exactly the same: here is the stuff that looks repeated:
```	
dat, err := ioutil.ReadFile(path)
	if err != nil {
		return errors.Wrap(err, "Failed to read config file: "+path)
	}

	config := ProfileConfig{Profiles: make(map[string]Profile)}
	if err = yaml.Unmarshal(dat, &config); err != nil {
		return errors.Wrap(err, "Failed to parse yaml file: "+path)
	}

	// get the correct profile
	var chosenProfile string
	if profileConfigKey == "" {
		chosenProfile = config.Default
	} else {
		chosenProfile = profileConfigKey
	}
```
I tried putting some of this in a function, and it doesn't work. The reason is that Yaml library needs specific type information on what it is reading into (ClusterConfig struct or ProfileConfigStruct). 
(**To be clear:** I tested this code and it does not work.)

```
func readYaml(path string, config interface{}) error {
	dat, err := ioutil.ReadFile(path)
	if err != nil {
		return errors.Wrap(err, "Failed to read config file: "+path)
	}

	if err = yaml.Unmarshal(dat, &config); err != nil {
		return errors.Wrap(err, "Failed to parse yaml file: "+path)
	}

}
```